src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.State.cs

```diff
@@ -110,6 +110,7 @@ internal sealed class State
             /// The <see cref="DateTime"/> values here do not update dynamically.
             /// </remarks>
             public IImmutableDictionary<string, DateTime> AdditionalDependentFileTimes { get; }
+            public DateTime LastAdditionalDependentFileTimesChangedAtUtc { get; }
```

Comment:
It would be good to add a comment to this property explaining why we need it. For example, this is actually the time at which the set of present items changed (not changing values).

The up to date check has some complex heuristics, so any information added here will likely be helpful in future.